incubator/hnc/hack/ci-test.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -e
+
+export PATH=$(go env GOPATH)/bin:$PATH
+mkdir -p $(go env GOPATH)/bin
+
+echo "Installing kubebuilder"
+wget https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.0.0-alpha.1/kubebuilder_2.0.0-alpha.1_linux_amd64.tar.gz
+tar -zxvf kubebuilder_2.0.0-alpha.1_linux_amd64.tar.gz
+mv kubebuilder_2.0.0-alpha.1_linux_amd64 /usr/local/kubebuilder
+echo "Finished installing kubebuilder"
+
+make test -C "${PWD}/incubator/hnc"
```

Comment:
Also run `make kubectl`and `make docker-build` since those are different targets not covered by `make test`.

Reply:
WDYT of creating a different file for this? One for the tests, which would be this one and another for these sort of preconditions, perhaps a `ci-check-preconditions.sh`. Then at the prow job we call these two files in two separate commands.

Reply:
I know that controller-runtime has a bunch of different files but I'm actually fine with leaving it as a single file for now. It's trivial to split it into multiple files in the future, so let's keep it easier to read for now.

Reply:
I'm not sure if I can run `make docker-build` inside a docker container :thinking:

Reply:
Can I just keep the `make test` and `make kubectl`?

Reply:
The section in that blog post called "The Solution" seems to be a reasonable solution - bind the Docker socket so that the other container gets created as a sibling. I'm assuming that the [`containers` list](https://github.com/kubernetes/test-infra/blob/4b879e817d005138c9775102a171f46054cb4303/config/jobs/kubernetes-sigs/controller-runtime/controller-runtime-presubmits.yaml#L12) is a list of [`Container`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#container-v1-core), which means you should be able to access the Docker socket like this: https://estl.tech/accessing-docker-from-a-kubernetes-pod-68996709c04b

Reply:
On the one hand, `make docker-build` is important because it tests the code inside of `Dockerfile`. Without that test, you could completely break the dockerfile and no-one would know.

On the other hand, it's fine to delay that to a later PR :) Maybe check in this "basic" PR now, then set up the Prow job (with access to the Docker socket), and if that's working, add `make docker-build` back to this script then.

Reply:
Oh, okay. I read the link you gave and it seems reasonable. I prefer to first do this simpler PR and then create another one with the docker build functionality.

Reply:
sgtm